### ISSUE TITLE
twisted.conch.scripts.cftp.StdioClient._printProgessBar blows up if the clock is too slow

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4463]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#4463|
|Type|enhancement|
|Created|2010-05-29 22:56:05Z|
|Branch|https://github.com/twisted/twisted/tree/no-progess-4463|

As seen in [tracebacks in buildbot logs](http://buildbot.twistedmatrix.com/builders/hardy32-py2.5-select/builds/93/steps/trial/logs/test.log), `_print`'''`Progess`'''`Bar` (sic) expects `startTime` to always be greater than `time.time()`.  It's apparently possible these can be equal (or maybe even in the opposite order, if the clock has been set!), so this function should take care not to divide by zero.

<details><summary>Searchable metadata</summary>

```
trac-id__4463 4463
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_no_progess_4463 branches/no-progess-4463
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__easy easy
time__1275173765000000 1275173765000000
changetime__1275239449000000 1275239449000000
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph removed owner<br>@glyph set status to `new`|
|-|-|

[Build Results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/no-progess-4463).

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_4"></a>|@mithrandi set owner to @glyph|
|-|-|

While you're changing the list of units, please correct the case of bytes and kilobytes (they should be 'B' and 'kB' respectively; 'b' is bits, and 'kb' is kilobits).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

(In [29092]) Capitalize "B" for "bytes" (as opposed to "b" for "bits") according to reviewer suggestion. re [#4463](https://github.com/twisted/twisted/issues/4463)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set status to `closed`|
|-|-|

(In [29093]) Merge no-progess-4463: fix a conch bug that was causing buildbot issues

Author: glyph

Reviewer: mithrandi

Fixes: [#4463](https://github.com/twisted/twisted/issues/4463)

The 'cftp' command-line tool will no longer encounter an intermittent error,
crashing at startup with a ZeroDivisionError while trying to report progress.
